### PR TITLE
feat: Allow any kind of AST fragment to be formatted

### PR DESCRIPTION
**Pull request by Marwes**
Adds `format_node` which accepts any AST fragment that `Node` can contain
and formats it. Should be useful for generating code fragments programmatically
(like that LSP wants to do) as this lets any AST be turned back into a string paving the way
for something akin to https://docs.rs/quote/latest/quote/macro.quote.html  (cc @rockstar ).

(I say string here, but it doesn't need to be a string per-se, as long as it is a common type
that the AST gets transformed to that may be passed around)

### Checklist

Dear Author :wave:, the following checks should be completed (or explicitly dismissed) before merging.

- [x] ✏️ Write a PR description, regardless of triviality, to include the _value_ of this PR
- [x] 🔗 Reference related issues
- [x] 🏃 Test cases are included to exercise the new code
- [x] 🧪 If **new packages** are being introduced to stdlib, link to Working Group discussion notes and ensure it lands under `experimental/`

Dear Reviewer(s) :wave:, you are responsible (among others) for ensuring the completeness and quality of the above before approval.

### CODE REVIEW

File: libflux/flux-core/src/formatter/mod.rs [hunk 1]
```diff
@@ -329,24 +346,84 @@ impl<'doc> Formatter<'doc> {
...
+            Node::PropertyType(x) => self.format_property_type(x),
+            Node::ParameterType(x) => self.format_parameter_type(x),
+            Node::TypeConstraint(x) => self.format_type_constraint(x),
+            Node::BadExpr(_) | Node::BadStmt(_) | Node::Package(_) => {
+                return Err(anyhow!("Unable to format node"))
+            }
+        })
+    }
+
+    fn format_package_clause(&mut self, pkg: &'doc ast::PackageClause) -> Doc<'doc> {
```
Comment (Marwes) [inline libflux/flux-core/src/formatter/mod.rs:400]: All changes from here on in this file are just boring extraction of code into functions so I could call them in `format_node`

File: libflux/flux-core/src/formatter/mod.rs [hunk 2]
```diff
@@ -52,12 +63,18 @@ pub struct Options {
     ///
     /// Default: true
     pub trailing_newline: bool,
+
+    /// How long a line is allowed to be before newlines are forced
+    ///
+    /// Default: 120
```
Comment (wolffcm) [inline libflux/flux-core/src/formatter/mod.rs:69]: There's been some internal discussion about line length as it affects accessibility, where `100` has been suggested.

Comment (wolffcm) [reply libflux/flux-core/src/formatter/mod.rs:69]: cc @pierwill

**Review by wolffcm — APPROVED** [review]
This looks good to me. Maybe change default line length to 100?

File: libflux/flux-core/src/formatter/mod.rs (hunk 2 above)
Comment (Marwes) [reply libflux/flux-core/src/formatter/mod.rs:69]: Sure, the only reason it was so high is that when I rewrote the formatter, having this line length kept the formatting that existed before (so less noise while getting things merged)

Comment (wolffcm) [reply libflux/flux-core/src/formatter/mod.rs:69]: Ah ok. Maybe we should change it to 100 as a separate issue.

**Pull request merged**